Adjustments for 5.4.2

joshmoore opened the pull request:
Various last-minute adjustments to the docs for the 5.4.2 release.

cc: @hflynn @pwalczysko @kennethgillen @jburel

joshmoore committed:
Updates to publication workflow

Follow on to gh-1815 in preparation for release of 5.4.2
as well as the related blog post.

joshmoore committed:
Minor refactoring to 'other resources' section

joshmoore committed:
Update whatsnew files

joshmoore committed:
Improve publication index

joshmoore commented:
Note: I'm concerned about the naming conflict between `public` and `publication`.

joshmoore committed:
Migrate publication to public

joshmoore committed:
Rework intro to complete workflow

joshmoore committed:
Fix whitespace

joshmoore committed:
Add link to prod-playbooks

jburel commented on omero/index.rst:
```diff
@@ -26,15 +26,15 @@ Additional online resources can be found at:
   for tutorials and presentations
 - `Demo server <http://qa.openmicroscopy.org.uk/registry/demo_account/>`_ -
-  now managed by the main OME team, providing the latest released version of
+  managed by the main OME team, providing the latest released version of
```
maybe also mention latest web apps?

joshmoore replied:
Pushed. Thoughts? Also, would be quite nice to see _some_ public data on demo. cc: @pwalczysko

joshmoore committed:
Modify layout for readability

- Remove bullets where not clearly "do 1, then 2"
- Add headers where missing
- Re-arrange sections

joshmoore committed:
Mention apps when it comes to the demo server

joshmoore commented:
Going to hold off on further `public(action).rst` changes until we've had a chance to discuss.

jburel commented on omero/sysadmins/public.rst:
```diff
@@ -138,3 +139,132 @@ authentication. For example a direct link to image will look as follows::
+of a complete workflow for using OMERO to host public data associated with a
+publication. It is illustrated using an example publication from the Swedlow
+lab in Dundee,
+`Schleicher et al, 2017 <http://dx.doi.org/10.1098/rsob.170099>`_ with the
```
update link to http://rsob.royalsocietypublishing.org/content/7/11/170099

joshmoore replied:
Isn't this precisely what DOIs are for?

jburel replied:
it is but elsewhere in the doc we use the "final" link
unifying makes the review easier

joshmoore replied:
I can't get that to work. Minimally those URLs use different subpaths: `/content/royopenbio` for jpegs vs. just `/content` for the target of the DOI.

jburel replied:
ignore in that case

joshmoore replied:
Looking for a "component DOI" now: http://blogs.plos.org/mfenner/2011/03/26/direct-links-to-figures-and-tables-using-component-dois/

kennethgillen replied:
@joshmoore - can we link ["launching web viewer"](https://github.com/openmicroscopy/ome-documentation/pull/1830/files#diff-f7e55dec15373a4b0fe85781b7e29929R13) back to this page?

joshmoore replied:
Does the link at the top count? (see https://github.com/openmicroscopy/ome-documentation/pull/1830/commits/c6cd42c9ae2602e4a2279522e86c6f346faeca2b)

kennethgillen replied:
A "Ref" with link directly after [the sentence "please remember"](https://github.com/openmicroscopy/ome-documentation/commit/c6cd42c9ae2602e4a2279522e86c6f346faeca2b#diff-b0666ac64875d77a655a2a545b172593R75) would be nicer imho.

joshmoore replied:
https://github.com/openmicroscopy/ome-documentation/pull/1830/commits/f9329233b51603d5d47d5bd8e407a1b937d81a14 ?

jburel commented on omero/sysadmins/public.rst:
```diff
@@ -138,3 +139,132 @@ authentication. For example a direct link to image will look as follows::
+data hosted at
+`<https://omero.lifesci.dundee.ac.uk/pub/schleicher-et-al-2017>`_.
+
+Ansible playbooks can be found describing how the "nightshade" server was
```
better to use ``how the production server in Dundee`` was ...

joshmoore replied:
I purposefully used "nightshade" because otherwise it isn't clear what playbooks to read in that repo. Otherwise, we need to rename the files.

jburel replied:
in that case maybe the suggested text and in brackets or i.e. "nightshade"

jburel commented on omero/sysadmins/public.rst:
```diff
@@ -138,3 +139,132 @@ authentication. For example a direct link to image will look as follows::
+A group-per-publication allows the public user to be selected added (or
+removed) from given publications to decide their visibility.
+
+#. Create a dedicated read-only group to host the raw data underlying the
```
dot missing at the end

joshmoore replied:
Updated following line to have a period.

joshmoore committed:
Suggestions from J-m

pwalczysko commented on omero/index.rst:
```diff
@@ -26,15 +26,15 @@ Additional online resources can be found at:
+  managed by the main OME team, providing the latest released versions of
+  OMERO and plugins for you to try out
+- OMERO API documentation - :javadoc:`OmeroJava API <>`,
+  :pythondoc:`OmeroPy API <>`,
```
I know that the target site for this link was not dealt with in this PR, but
![screen shot 2018-01-19 at 10 18 26](https://user-images.githubusercontent.com/2478303/35146199-2ff7a4aa-fd02-11e7-9ecb-7e3840a6e1ff.png)
looks strange. Is this intended ?

joshmoore replied:
I think that's mostly if not completely a result of the `latest/` redirect.

hflynn replied:
It is yes, should be fine once the release is live, I checked the conf.py just now

pwalczysko commented on omero/sysadmins/index.rst:
```diff
@@ -137,8 +136,11 @@ OMERO clients to host images for groups or public viewing.
+    :maxdepth: 2
+    :titlesonly:
+
+    customization
```
![screen shot 2018-01-19 at 10 35 13](https://user-images.githubusercontent.com/2478303/35146953-aefb8562-fd04-11e7-80f8-8f625ff534eb.png)
This seems tp be a strange order. First I am told about 2 items which are (at least highlighted as) particular, only then about an item which claims to be "complete workflow".

joshmoore replied:
I went back and forth on this quite a bit. The introduction to the workflow, though, links to public_user making it something of a prerequisite. Additionally, "complete workflow" doesn't show everything in the other two sections. So options I can think of:

 * Integrate "complete workflow" into the rest of the page using the single example over and over again (high cost, but I think valuable)
 * Append "Setup:" or "Howto:" to the preliminary items
 * Other names for "Complete workflow"
    - "Full example for ..."
    - "Motivation: ..."

pwalczysko replied:
I guess the attack on the "Complete workflow" words is the easiest and would sort 90% of what I mean. I am afraid dispersing the content completely would create problems for simple linking from the blog post.

``Full example of ... `` would sound best to me

joshmoore replied:
Pushed

will-moore commented on omero/sysadmins/public.rst:
```diff
@@ -138,3 +139,133 @@ authentication. For example a direct link to image will look as follows::
+Ansible playbooks can be found describing how the production server in Dundee
+("nightshade") was configured in the
+`prod-playbooks <https://github.com/openmicroscopy/prod-playbooks>`_
+repository on GitHub.
```
Do we want to indicate which playbooks we're referring to here, since there are a few at that repo?

joshmoore replied:
There are several files that are related. If we're ok with expanding this section substantially, then can do. cc: @kennethgillen @manics

will-moore replied:
Maybe just add that info to the repo's README, instead of here.

joshmoore replied:
See https://github.com/openmicroscopy/prod-playbooks/pull/30

manics replied:
Just in case you weren't aware a lot of the important config (e.g. omero.web.public config) is currently in private group_vars.

joshmoore replied:
Yeah, I'm realizing having tried to get demo running in molecule:

```
+ansible:
+  playbook: playbook.yml
+  host_vars:
+    ome-demoserver.openmicroscopy.org:
+      # Override due to testing environment
+      omero.jvmcfg.system_memory: 2048
+      # Variables which are assumed to exist (missing defaults)
+      omero_server_mail_from: root@localhost
+      omero_server_mail_host: localhost
+      omero_web_admins: [[ "Sysadmin", "root@localhost"]]
+      nginx_ssl_files_path: /etc/nginx/ssl
+      vault:
+        omero_server_db_user: omero
+        omero_server_dbpassword: password
+        omero_server_dbname: omero
+        omero_server_rootpassword: password
+        omero_web_public_user: public_user
+        omero_web_public_password: password
```

I'm going to propose that we put defaults for all of these in the repo and only override what's truly secret.

manics replied:
This is what I've done for the IDR:
- default/secret password: https://github.com/IDR/deployment/blob/IDR-0.4.4/ansible/group_vars/omero-hosts.yml#L123
- https cert location: https://github.com/IDR/deployment/blob/IDR-0.4.4/ansible/group_vars/proxy-hosts.yml#L15
- https self-signed control https://github.com/IDR/deployment/blob/IDR-0.4.4/ansible/idr-proxy.yml#L87

hflynn commented:
Updated https://github.com/ome/blog/pull/50 to reflect the new URL for the workflow, will update the website PR next.

pwalczysko commented on omero/sysadmins/public.rst:
```diff
@@ -138,3 +139,133 @@ authentication. For example a direct link to image will look as follows::
+Group setup
+^^^^^^^^^^^
+
+A group-per-publication allows the public user to be selected added (or
```
``...selected added (or removed)...`` ?
Maybe ``to be added (or removed)`` or at least put comma between ``selected`` and ``added`` ?

joshmoore replied:
I think I meant "selectively". Pushing that now.

pwalczysko commented on omero/sysadmins/public.rst:
```diff
@@ -138,3 +139,133 @@ authentication. For example a direct link to image will look as follows::
+will need to be specially configured to permit loginless access. This example
+uses a second, dedicated server.
+
+#. Configure the :ref:`public_user`.
```
Unfortunately, this reads now ``Configure the Configuring public user.`` when built

joshmoore replied:
Pushed

joshmoore committed:
Fix selectively typo

joshmoore committed:
Fix double periods

will-moore commented on omero/sysadmins/public.rst:
```diff
@@ -138,3 +139,133 @@ authentication. For example a direct link to image will look as follows::
+   :doc:`separate web server <unix/install-web/web-deployment>` for servicing
+   the public workflows can be added and configured to point at your existing
+   OMERO.server (in this case, the public user and the public URL filter need
+   to be set on this separate OMERO.web installation).
```
It might be better to discuss the logic / decision-making for setting up a second web server as a first step *before* the config steps. Currently it looks like you can make this decision at point 3, after doing steps 1 and 2.

E.g.
"Configuring OMERO.web

If you wish to have an automatically logged-in public user while still giving your existing OMERO users an unchanged user experience (i.e. not automatically logging them in as the public user), a dedicated, separate web server for servicing the public workflows can be added and configured to point at your existing OMERO.server. 
This is the workflow we adopted by adding a public OMERO.web at omero.lifesci.dundee.ac.uk, without changing the existing OMERO.web at nightshade.openmicroscopy.org.

#. Create the public user as described above. 
#. Configure the public user settings for OMERO.web, setting the :property:`omero.web.public.url_filter` to allow access to the full webclient UI as :ref:`described <public.url_filter>`.

joshmoore replied:
Pushed

pwalczysko commented on omero/sysadmins/whatsnew.rst:
```diff
@@ -10,6 +10,9 @@ What's new for OMERO 5.4 for sysadmins
 - A walkthrough has been added for :doc:`unix/server-debian9-ice36`.
 
+- A walkthrough has been added to :doc:`public`, based on a concrete
+  use-case in Dundee.
```
As there is no summary list of links on the ``Publishing data using OMERO.web`` page (unlike on the other pages to which the What's new points) I would add some hint about how to look for it on the page. Something like the name of the paragraph of the walkthrough itself. This will of course depend on how the paragraph will be named as discussed in https://github.com/openmicroscopy/ome-documentation/pull/1830/files#r162590029

pwalczysko commented:
Finished reviewing. Couple of minor comments.

joshmoore committed:
Add link after 'user can access'

joshmoore committed:
Fix webclient property capitalization

joshmoore committed:
Change title to 'Full example'

joshmoore committed:
Fix 'Configuring public_user' link

joshmoore commented:
> Not sure about the capitalization or not

@pwalczysko : pushed

joshmoore committed:
Rework configuring web section

joshmoore committed:
Link explicitly to the walkthrough

hflynn commented:
This all looks good to me, thanks very much. Having the subheadings for this page on the sysadmin index page is definitely valuable and the reworked page flows nicely.

I'm happy to merge if there are no further comments. Are you happy now @pwalczysko ?

will-moore commented:
Looks good, thanks.

pwalczysko commented:
Yes, it reads well. All problems fixed. Thank you.

joshmoore commented:
Thanks everyone.

kennethgillen commented on omero/sysadmins/public.rst:
```diff
@@ -138,3 +139,131 @@ authentication. For example a direct link to image will look as follows::
+link to the correct group and Project in OMERO, just as the
+DOI above does. Redirects need to be set in the
+`NGINX <http://nginx.org/>`_ component of the OMERO.web installation
+dedicated to publication workflows.
```
@joshmoore  - worth linking to our example at https://github.com/openmicroscopy/prod-playbooks/blob/master/nightshade-webclients.yml#L172 ?

hflynn commented:
I'll open a PR for Kenny's suggestion now
